java/examples/XSSCustomSanitizer.ql

```diff
@@ -0,0 +1,48 @@
+/**
+ * @name Cross-site scripting
+ * @description Writing user input directly to a web page
+ *              allows for a cross-site scripting vulnerability.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 6.1
+ * @precision high
+ * @id java/xss
+ * @tags security
+ *       external/cwe/cwe-079
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.security.XSS
+import DataFlow::PathGraph
+
+class CustomXSSSanitizer extends XssSanitizer {
+  CustomXSSSanitizer() {
+    this.asExpr()
+        .(MethodAccess)
+        .getMethod()
+        .hasQualifiedName("org.example", "Sanitizes",  ["escapeHtml", "escapeJavaScript"])
+  }
+}
+
+class XSSConfig extends TaintTracking::Configuration {
+  XSSConfig() { this = "XSSConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof XssSink }
+
+  // Add our custom sanitizer class in the configuration
```

Comment:
This comment can probably be removed to avoid confusing the customer, as the TaintTracking Configuration is the same one of the original query.